### PR TITLE
fix(web-components): open ic-dialog during initialisation using the open prop

### PR DESCRIPTION
**Pull request by dz333dz**
## Summary of the changes
A new open prop has recently been added to the IcDialog component, which replaces the deprecated showDialog and hideDialog methods.

The trouble is, when an instance of an IcDialog component is initialised using the new open prop, with open="true", the dialog remains hidden. This is because the dialog is only made visible when the value of the open prop changes, i.e. when it is assigned a different value after the component has been initialised (which happens in the watchOpenHandler method).

In particular, this causes a problem for testing IcDialog based components using Cypress component tests, as it is currently not possible to instantiate a component using a prop of open="true" and then just go ahead and run assertions in Cypress, as a change of state is needed for the open prop to make the dialog visible in the tests, which involves some hacky and unreliable workarounds to get it working.

The small change in this PR fixes the issue. Now, if users of the IcDialog component pass in open="true", then the dialog is made visible during initialisation within the componentDidLoad function, reusing the existing code that executes whenever the watchOpenHandler is executed.

## Related issue
N/A

## Checklist
- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] All acceptance criteria reviewed and met. 
- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.
- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] Browser support tested (Chrome, Safari, Firefox and Edge).
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 
- [x] All prop combinations work without issue. 
- [x] Changes to docs package checked and committed.

### CODE REVIEW

File: packages/web-components/src/components/ic-dialog/ic-dialog.tsx [hunk 1]
```diff
@@ -117,46 +117,17 @@ export class Dialog {
...
+      this.dialogOpened();
     } else {
       this.fadeIn = false;
       if (this.resizeObserver !== null) {
         this.resizeObserver.disconnect();
       }
       setTimeout(() => {
         this.dialogRendered = false;
-        this.dialogEl.close();
+        if (this.dialogEl?.close) {
```
Comment (jd239) [inline packages/web-components/src/components/ic-dialog/ic-dialog.tsx:128]: I do not think this if statement is required as `dialogEl` is always going to be a `<dialog>` element and the close method comes standard with it. 
ref: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog

**Review by jd239 — CHANGES_REQUESTED** [review]
Nice addition to the dialog component. Thanks for the contribution. 🚀 
A few things:
- Could you please create the 'Auto Open' story for the Web Components storybook

- The commits need to be separated based on the update to the package so for example:
   - The web component commit scope should only contain the update to dialog.tsx and the new web component story
   - Docs commit for the update to docs.json
   - React commit for React storybook

Hopefully this helps: https://github.com/mi6/ic-ui-kit/blob/main/CONTRIBUTING.md#git-commit

Thanks.

File: packages/web-components/src/components/ic-dialog/ic-dialog.tsx (hunk 1 above)
Comment (dz333dz) [reply packages/web-components/src/components/ic-dialog/ic-dialog.tsx:128]: Agreed - this was left over from trying to getting this to work using a different lifecycle method. I will revert this change.

**Comment by dz333dz** [comment]
> Nice addition to the dialog component. Thanks for the contribution. 🚀 A few things:
> 
> * Could you please create the 'Auto Open' story for the Web Components storybook
> * The commits need to be separated based on the update to the package so for example:
>   
>   * The web component commit scope should only contain the update to dialog.tsx and the new web component story
>   * Docs commit for the update to docs.json
>   * React commit for React storybook
> 
> Hopefully this helps: https://github.com/mi6/ic-ui-kit/blob/main/CONTRIBUTING.md#git-commit
> 
> Thanks.

Apologies, I missed the fact that there are 2 different storybooks to update! I will make sure that the "Auto opening" (as well as the "Hidden close button" story that I recently added) appears in both storybooks.

I will also look into making the changes in separate commits, as suggested.

**Comment by dz333dz** [comment]
All comments addressed - ready for re-review.

**Review by GCHQ-Developer-530 — CHANGES_REQUESTED** [review]
Hi @dz333dz thank you for your contribution! Looks really good, just one small change. Please could you update both storybook examples to have a button to reopen the dialog after it has been closed? At the moment it just leaves the blank page as there's nothing there to reopen it. Thank you

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Pull request merged**